OBSDOCS-1726: Port the "Installing log storage" sections to 5.8 and 6… (#95028)
openshift/openshift-docs: configuring/configuring-the-log-store.adoc

:_mod-docs-content-type: ASSEMBLY
include::_attributes/common-attributes.adoc[]
[id="configuring-lokistack-storage"]
= Configuring the log store

> :context: configuring-the-log-store

Reviewer: Must have an empty line between H1 heading and first paragraph

toc::[]

You can configure a `LokiStack` custom resource (CR) to store application, audit, and infrastructure-related logs.

include::snippets/loki-statement-snip.adoc[leveloffset=+1]

// Loki sizing
include::modules/loki-sizing.adoc[leveloffset=+1]

// Loki object storage
include::modules/logging-loki-storage.adoc[leveloffset=+1]

// create object storage
include::modules/logging-loki-storage-aws.adoc[leveloffset=+2]
include::modules/logging-loki-storage-azure.adoc[leveloffset=+2]
include::modules/logging-loki-storage-gcp.adoc[leveloffset=+2]
include::modules/logging-loki-storage-minio.adoc[leveloffset=+2]
include::modules/logging-loki-storage-odf.adoc[leveloffset=+2]
include::modules/logging-loki-storage-swift.adoc[leveloffset=+2]

//Loki stirage STS
[id="installing-log-storage-loki-sts"]
=== Deploying a Loki log store on a cluster that uses short-term credentials

> 

Reviewer: No level 3 headings (===) in any files

Author: this will be done as part of contentx because we have many instances of third level headings in the doc and changing the heading level here involves a lot of restructuring.

For some storage providers, you can use the Cloud Credential Operator utility (`ccoctl`) during installation to implement short-term credentials. These credentials are created and managed outside the {ocp-product-title} cluster. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/authentication_and_authorization/managing-cloud-provider-credentials#cco-short-term-creds[Manual mode with short-term credentials for components].

[NOTE]
====
Short-term credential authentication must be configured during a new installation of {loki-op}, on a cluster that uses this credentials strategy. You cannot configure an existing cluster that uses a different credentials strategy to use this feature.
====

include::modules/logging-identity-federation.adoc[leveloffset=+3]

include::modules/logging-create-loki-cr-console.adoc[leveloffset=+3,tag=!pre-5.9]

include::modules/loki-create-object-storage-secret-cli.adoc[leveloffset=+3]

include::modules/logging-loki-log-access.adoc[leveloffset=+2,tag=!NetObservMode]

> 

Reviewer 2: Delete this extra line.

Author: addressed here: https://github.com/openshift/openshift-docs/pull/95640

include::modules/logging-creating-new-group-cluster-admin-user-role.adoc[leveloffset=+2]

//Enhanced reliability and performance
[id="performance_{context}"]

> == Enhanced reliability and performance

Reviewer 2: In case someone lands on this heading through a search, it might be wise to make the heading more specific. For example, "Enhanced reliability and performance for Loki".

Use the following configurations to ensure reliability and efficiency of Loki in production.

include::modules/loki-pod-placement.adoc[leveloffset=+2]

include::modules/logging-loki-reliability-hardening.adoc[leveloffset=+2]

include::modules/loki-retention.adoc[leveloffset=+2]

> 

Reviewer 2: Delete this extra line.

Author: done in https://github.com/openshift/openshift-docs/pull/95640

include::modules/loki-memberlist-ip.adoc[leveloffset=+2]

include::modules/loki-restart-hardening.adoc[leveloffset=+2]

//Advanced deployment and scalability
[id="advanced_{context}"]

> == Advanced deployment and scalability

Reviewer 2: Similar suggestion here to use a more specific heading to help people who find this section through search. Maybe something like, "Advanced deployment and scalability for Loki"?

Author: https://github.com/openshift/openshift-docs/pull/95640

> To configure high availability, scalability, and error handling, use the following information.

Reviewer 2: ```suggestion
You can configure high availability, scalability, and error handling for Loki.
```

include::modules/loki-zone-aware-replication.adoc[leveloffset=+2]
include::modules/loki-zone-fail-recovery.adoc[leveloffset=+2]
include::modules/loki-rate-limit-errors.adoc[leveloffset=+2]

//log based alerts
[id="log-based-alerts_{context}"]

> == Log-based alerts

Reviewer 2: Similar comment here about adjusting the heading to be more specific. Maybe something like, "Log-based alerts for Loki"?

> 

Reviewer 2: Be sure to add a short introduction here that explains why someone might want to know this information.
